crossbeam-epoch/src/sync/queue.rs

```diff
@@ -137,17 +134,20 @@ impl<T> Queue<T> {
         let head = self.head.load(Acquire, guard);
         let h = unsafe { head.deref() };
         let next = h.next.load(Acquire, guard);
-        match unsafe { next.as_ref() } {
-            Some(n) if condition(&n.data) => unsafe {
-                self.head
-                    .compare_and_set(head, next, Release, guard)
-                    .map(|_| {
-                        guard.defer_destroy(head);
-                        Some(ManuallyDrop::into_inner(ptr::read(&n.data)))
-                    })
-                    .map_err(|_| ())
-            },
-            None | Some(_) => Ok(None),
+        unsafe {
```

Comment:
It seems the scope of `unsafe` is enlarged---why?

Reply:
Because the if guard of the match requires an unsafe dereference, so that would require an unsafe scope, however it would rather seem "weird" applying unsafe for just some fragments for this match block, although that would be more precise. I decided against precision and opted for a more cosmetic approach, since really most of the things in this match block is unsafe: `as_ref`,  dereferencing raw pointer, the matching `match` block.

Reply:
I prefer the original version, because it is accurately describing the fact that (1) converting a raw pointer to reference is potentially unsafe, and (2) we manually inspected (at least, in theory...) the safety of the conversion.  The conversion is safe because we have a sufficient permission to (de)reference the node of our interest after acquire load.

I believe this discussion can be made into another PR.  How about merging this PR without this change for now?

Reply:
I personally don't really mind, so pushed a new commit which reverts this :)